src/containers/welcome/LoginModal.tsx

```diff
@@ -0,0 +1,134 @@
+import { Modal, Form, Alert } from "react-bootstrap";
+import { useNavigate } from "react-router-dom";
+import { useState } from "react";
+import styled from "styled-components";
+import { RoutePaths } from "../../common/constants/routes";
+
+interface LoginModalProps {
+  show: boolean;
+  handleClose: () => void;
+}
+
+const ModalHeader = styled(Modal.Header)`
+  background-color: ${({ theme }) => theme.accentBackgroundColor};
+  color: ${({ theme }) => theme.blue};
+  border: transparent;
+  border-top-left-radius: 15px;
+  border-top-right-radius: 15px;
+`;
+
+const HeaderText = styled(Modal.Title)`
+  font-size: 30px;
+  margin-left: 20px;
+`;
+
+const ModalBody = styled(Modal.Body)`
+  background-color: ${({ theme }) => theme.accentBackgroundColor};
+  border-bottom-left-radius: 15px;
+  border-bottom-right-radius: 15px;
+`;
+
+const StyledModal = styled(Modal)`
+  .modal-content {
+    border-radius: 15px;
+    border: transparent;
+  }
+`;
+
+const UsernameStyle = styled(Form.Control)`
+  background-color: ${({ theme }) => theme.disabledBackground};
+  border-style: solid;
+  border-width: 2px;
+  border-color: ${({ theme }) => theme.blue};
+`;
+
+const PasswordStyle = styled(Form.Control)`
+  background-color: ${({ theme }) => theme.disabledBackground};
+  margin-top: 20px;
+  border-style: solid;
+  border-width: 2px;
+  border-color: ${({ theme }) => theme.blue};
+`;
+
+const Button = styled.button<{ $login?: boolean }>`
+  background-color: ${(props) =>
+    props.$login ? props.theme.green : props.theme.green};
+  color: ${(props) => (props.$login ? props.theme.backgroundColor : props.theme.backgroundColor)};
+  font-weight: bold;
+  border: 2px solid ${(props) => (props.$login ? props.theme.green : props.theme.green)};
+  width: 50%;
+  margin-top: 10px;
+  border-radius: 20px;
+  padding: 0.5em 1em;
+  cursor: pointer;
+  margin: 10px auto;
+  display: block;
+
+  &:hover {
+    background-color: ${(props) =>
+      props.$login ? props.theme.accentBackgroundColor : props.theme.accentBackgroundColor};
+    color: ${(props) => (props.$login ? props.theme.green : props.theme.green)};
+    border-color: ${(props) => props.theme.green};
+    border-width: 2px;
+  }
+`;
+
+export function LoginModal({ show, handleClose }: LoginModalProps) {
+  const navigate = useNavigate();
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+  const [error, setError] = useState("");
+
+  const storedUsername = import.meta.env.VITE_USERNAME || "admin";
+  const storedPassword = import.meta.env.VITE_PASSWORD || "admin";
```

Comment:
since you are storing the user data in the local storage when the user signs up, why not check if the credentials are present in the local storage and allow the user to login using those credentials as well?